doc/user/path.rst

```diff
@@ -0,0 +1,415 @@
+.. _path:
+
+****
+PATH
+****
+
+:abbr:`PATH` is a daemon that handles the installation and deletion
+of Segment Routing (SR) Policies.
+
+
+.. _starting-path:
+
+Starting PATH
+=============
+
+Default configuration file for *pathd* is :file:`pathd.conf`.  The typical
+location of :file:`pathd.conf` is |INSTALL_PREFIX_ETC|/pathd.conf.
+
+If the user is using integrated config, then :file:`pathd.conf` need not be
+present and the :file:`frr.conf` is read instead.
+
+.. program:: pathd
+
+:abbr:`PATH` supports all the common FRR daemon start options which are
+documented elsewhere.
+
+
+PCEP Support
+============
+
+To build the PCC for pathd, the `pceplib external library <https://github.com/volta-networks/pceplib>`_.
+The last commit of pceplib the module has been tested with is stored in
+the file `pathd/pceplib.lock`.
+The following extra parameter should be passed to `configure`:
+
+```
+--enable-pcep LDFLAGS="-L${PCEPLIB_ROOT}/install/lib" CPPFLAGS="-I${PCEPLIB_ROOT}/install/include"
+```
+
+To start pathd with pcep support the extra parameter `-M pathd_pcep` should be
+passed to the pathd daemon.
+
+
+.. _path-commands:
+
+
+Pathd Commands
+==============
+
+Configuration example:
+
+.. code-block:: frr
+
+  debug pathd pcep basic
+  pcep-config-group GROUP1
+   source-address 1.1.1.1
+   tcp-md5-auth secret
+   timer keep-alive 30
+  !
+  pcc-peer PCE1
+   config-group GROUP1
+   address ip 10.10.10.10
+  !
+  pcc-peer PCE2
+   config-group GROUP1
+   address ip 9.9.9.9
+  !
+  pcc
+   peer PCE1 precedence 10
+   peer PCE2 precedence 20
+  !
+  segment-list SL1
+   index 10 mpls label 16010
+   index 20 mpls label 16030
+  !
+  sr-policy color 1 endpoint 1.1.1.1
+   name default
+   binding-sid 4000
+   candidate-path preference 100 name CP1 explicit segment-list SL1
+   candidate-path preference 200 name CP2 dynamic
+    affinity include-any 0x000000FF
+    pcep bandwidth 100000
+    pcep required metric bound msd 16
+    pcep desired metric te 10
+    pcep required objective-function mcp mlp mlp
```

Comment:
- We used `pcep` because it wasn't clear that these parameters where generic enough, for example, the metric names are right out of the pcep specification.
 - About using `metric-te` I don't quite understand what you mean. The name could be changed to `te-metric` but then the command would be `pcep metric te-metric` as `metric` is used to differentiate the command that set metrics from the on setting bandwidth (for example).
 - About removing desired/required, I don't see why it should. This controls the pcep P flag, and controlling that is required by Volta. If this wasn't specified like this, how would you control the PCEP P flag from the configuration ?

Reply:
You should made a distinction between the PCEP protocol (the how) from what you are really configured here i.e. a segment path. So, even if the request is convey by PCEP, the parameters are linked to the segment path. If you have no PCEP session, we could imagine that pathd embed some path computation algorithms able to compute the path of the segment based on the segment constraints. So, all of these constraints must be under a `segment-path` header instead of `pcep` header. Cisco use its `tunnel` semantic and Juniper a `source-routing-path` one.

Regarding, required/desired, both Cisco and Juniper leave the P flag unset. Thus, by default you must set the desired for a P flag = 0 and I think it is not necessary to specify the default value. Adding only a `required` item at the end of all commands is largely sufficient.

Finally, for constraints, the `metric` keyword is superfluous. Again, Cisco and Juniper used directly the name of the metric for the constraints.

Reply:
If I understand correctly, you mean that the bandwidth command should be out of the pcep sub-command, but metrics and objective functions are fine ? metric is there to differentiate between metrics and objective functions. the other option would be to add a command per metric but that would be way too much command declarations in FRR.